lib/gnat/parsec.ex

```diff
@@ -0,0 +1,104 @@
+defmodule Gnat.Parsec do
+  defstruct partial: nil
+
+  import NimbleParsec
+
+  subject = ascii_string([?!..?~], min: 1)
+  length = integer(min: 1)
+  sid = integer(min: 1)
+  whitespace = ascii_char([32, ?\t])
+               |> times(min: 1)
+  op_msg = ascii_char([?m,?M])
+           |> ascii_char([?s,?S])
+           |> ascii_char([?g,?G])
```

Comment:
Curiosity: are those either "MSG" or "msg" or can they be literally "MsG"?

Reply:
That [nats documentation](https://nats.io/documentation/internals/nats-protocol/#protocol-messages) says:

> NATS protocol operation names are case insensitive, thus `SUB foo 1\r\n` and `sub foo 1\r\n` are equivalent

So I went with the literal "case-insensitive" definition, but their example only shows an upcase or downcase 🤷‍♂️